inputgraph/loader.go

```diff
@@ -216,25 +230,17 @@ func (l *loader) expandDirs(dirs ...string) ([]string, error) {
 	return uniqStrs(ret), nil
 }
 
-func uniqStrs(all []string) []string {
-	m := map[string]struct{}{}
-	for _, v := range all {
-		m[v] = struct{}{}
-	}
+func (l *loader) expandArgs(ctx context.Context, args []string) []string {
 	ret := []string{}
-	for k := range m {
-		ret = append(ret, k)
+	for _, arg := range args {
+		expanded := l.varCollection.ExpandOld(arg) // TODO: correct func? need to shell out?
```

Comment:
I'm using the `ExpandOld` method because `Expand` seemed to be for shell-out expansion. Is this correct?

Reply:
I would use `Expand`, and pass a function that prevents shelling out, you can follow the pattern used in https://github.com/earthly/earthly/blob/410c690e4df488287225c713da16d817a3488932/earthfile2llb/converter.go#L1710

and use something like:

```
	return c.varCollection.Expand(word, func(cmd string) (string, error) {
		return "", errShellOutNotPermitted
	})
```

Reply:
Thanks, I'll give that a shot.